device/common/include/traccc/finding/device/apply_interaction.hpp

```diff
@@ -36,7 +36,7 @@ struct apply_interaction_payload {
      * @brief View object to the vector of boolean-like integers describing
      * whether each parameter is live. Has the same size as \ref params_view
      */
-    vecmem::data::vector_view<const unsigned int> params_liveness_view;
```

Comment:
Is there a reason for this change? CUDA doesn't natively support 8-bit loads so I'm a bit worried about the performance implications of this. Also is there a reason to use `char` and not `unsigned char`?

Reply:
We use these values as bools. Using 32-bits where we only need 1, seems very silly.

Of course `bool` doesn't work. 😦 But our convention in the offline code is to use `char` when we need "boolean information", but `bool` can't be used.

Also, take this into account: https://github.com/acts-project/traccc/blob/main/device/cuda/src/finding/finding_algorithm.cu#L154

What do you think is actually getting set for this buffer with that operation? 😏 Because it's not `0x1` values in the `unsigned int` variables...

In any case, I can't see why we shouldn't go for this. Even if NVIDIA always copies at least 16 bits, right now we move 32 bits in all cases. Even though we only need 1. If some of the loads are next to each other, this could still win us a little bit.

Reply:
> What do you think is actually getting set for this buffer with that operation? 😏 Because it's not 0x1 values in the unsigned int variables...

It's setting a non-zero value; I don't see the problem?

Reply:
> In any case, I can't see why we shouldn't go for this. Even if NVIDIA always copies at least 16 bits, right now we move 32 bits in all cases. Even though we only need 1. If some of the loads are next to each other, this could still win us a little bit.

Think of what happens if four threads want to write their unsigned chars to global memory at the same time.

I seriously do not see why we need to change this to save what boils down to 240 kilobytes...

Reply:
Logically the code worked. But you can't argue that we're not spending more time with memsetting these values, and using more global memory on it, then we need to. 🤔